float32 width
---
bool success
string message
